File: apps/api/models/skill.go
package models

// NodeType represents the kind of skill node (API, CLI, or BROWSER).
const (
	NodeTypeAPI     = "API"
	NodeTypeCLI     = "CLI"
	NodeTypeBROWSER = "BROWSER"
)

// SkillNode is a single node in the skill graph.
type SkillNode struct {
	ID        string                 `json:"id"`
	Type      string                 `json:"type"` // API | CLI | BROWSER
	Config    map[string]interface{} `json:"config"`
	NextNodes []string               `json:"next_nodes"`
}

// SkillManifest is the full skill-set with PQC verification proof.
type SkillManifest struct {
	Name      string      `json:"name"`
	Version   string      `json:"version"`
	Nodes     []SkillNode `json:"nodes"`
	PQCProof  string      `json:"pqc_proof,omitempty"`
	SignedBy  string      `json:"signed_by,omitempty"` // e.g. "Vericore" for Verified Creator badge
}

File: apps/web/src/types/skill.ts
// Mirrors apps/api/models/skill.go for type-safe API usage.

export const NODE_TYPES = ["API", "CLI", "BROWSER"] as const;
export type NodeType = (typeof NODE_TYPES)[number];

export interface SkillNodeData extends Record<string, unknown> {
  id: string;
  type: NodeType;
  config: Record<string, unknown>;
  label?: string;
}

export interface SkillNodePayload {
  id: string;
  type: NodeType;
  config: Record<string, unknown>;
  next_nodes: string[];
}

export interface SkillManifest {
  name: string;
  version: string;
  nodes: SkillNodePayload[];
  pqc_proof?: string;
  signed_by?: string;
}

export interface ValidateResult {
  valid: boolean;
  errors?: string[];
}
